client/v3: do not overwrite authTokenBundle on dial

awly opened the pull request:
`Client.dial` can be called multiple times. For example, from regular
`NewClient` and [from the `Maintenance`
wrapper](https://github.com/etcd-io/etcd/blob/6d451ab61d2da992622b3e7dbf85ded04ebcc515/client/v3/maintenance.go#L81-L84).

This ends up creating two (if `Client.dial` is called twice) independent
`credentials.Bundle` instances:
- the first one is wired into `PerRPCCredentials` callback in gRPC
  client.
- the second one is in `Client.authTokenBundle` and receives the latest
  auth token updates.

When using username/password authentication and the server is configured
to issue JWTs, token rotation logic breaks because of the above
`credentials.Bundle` confusion.

As a simple solution, do not overwrite `Client.authTokenBundle` if it's already set.

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

ptabor reviewed (approved):
Thank you. Could you, please, add a regression test for the new behavior.

awly commented:
@ptabor sorry for the delay, added a test.
Also realized that `authTokenBundle` should really just be created in `newClient` instead of `dial`, moved it there.

awly commented:
@ptabor could you merge this please?

biosvs commented:
@ptabor how do you think, is it possible to make a backport to 3.5 for this fix? For cluster with enable auth this bug is quite annoying and hard to workaround without hot patching.
